### DISCUSSION TITLE
Problema en cobertura de código en los tests, Java Spring Boot

### DISCUSSION THREAD

**Discussion by Pelayori** [Arquisoft/faq#62] [Q&A] [answered]
A la hora de ejecutar los tests para que SonarCloud analize todo lo necesario, cobertura incluida, solo se tiene en cuenta el reporte de la clase de tests de los tests unitarios. La cobertura de los tests de integración no se tiene en cuenta.
Tenemos 2 clases de tests, una para unitarios y otra para integración.
Hemos probado de todo lo que hemos encontrado en internet, y nada ha funcionado. 
Por ejemplo, al ejecutar los tests, todo el código cubierto en los unitarios se ve reflejado en sonar, pero el código cubierto por lo de integración, principalmente controladores, quedan como "no cubiertos".

Si alguien tiene algo de ayuda, sería de agradecer. 
Nuestro pom.xml se puede consultar aquí: https://github.com/Arquisoft/wiq_es04b/blob/develop/pom.xml

**Comment by pglez82** [comment]
Buenos días,

Por defecto, como está configurado wiq_0, solo la cobertura de los tests unitarios se manda a sonarcloud. 

Ya he visto que tenéis todo el proyecto en Java. El problema es que yo al menos no estoy muy puesto en esta tecnología. Aún así, yo realizaría un par de comprobaciones para diagnosticar el problema:

1) Aseguraros de que el actions está corriendo los tests de integración. Para eso, tenéis que mirar en el log del actions para ver que se está ejecutando lo que se debe. Entiendo que queréis ejecutar tests unitarios y de integración usando este actions (https://github.com/Arquisoft/wiq_es04b/blob/develop/.github/workflows/unit-tests-push.yml). En el log de la última ejecución de este action que tenéis (https://github.com/Arquisoft/wiq_es04b/actions/runs/8526475843/job/23355806410) yo creo que solo se está ejecutando un test (no sé si esto es correcto). Revisar.
2) En lugar de probar todo el sistema de integración continua, podéis probar a ejecutar los tests de integración en local y probar a subir el reporte a sonarcloud desde vuestro ordenador usando el cliente de sonar-scanner. Así os quitáis toda la configuración del CI de delante y podéis probar la configuración de sonarcloud directamente.

Espero que las indicaciones os sirvan de algo.

**Reply by Pelayori** [reply]
Gracias, probaremos en local

**Comment by Pelayori** [comment, marked as answer]
Al final se ha solucionado. La clave está en el que java agent que se acopla a la aplicación desplegada para analizar la cobertura, no genera el reporte hasta que la aplicación se ha apagado. Una vez se ha apagado, el reporte de cobertura se genera, y posteriormente se ejecuta la goal de report del plugin de jacoco para maven.
Una vez hecho esto el reporte xml generado se puede enviar a sonar cloud y ahí ya analiza todo debidamente.
